### PR TITLE
Feature/orca 557 Point to GraphQL prototype for internal_reconcile_report_phantom API

### PR DESCRIPTION
**Pull request by AndrewDorn**
## Summary of Changes

Please write a sentence or two summarizing the proposed change.

Addresses [ORCA-557: Point to GraphQL prototype for internal_reconcile_report_phantom API](https://bugs.earthdata.nasa.gov/browse/ORCA-557)

## Changes

* Added documentation for Phantom API in GraphQL
* Removed GraphQL feature flag and updated customer documentation
* Typing bugfixes around GraphQL int definition

## Type of change

- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [x] This change requires a documentation update

## How Has This Been Tested?

Deployed to and tested in AWS.

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the necessary documentation (remove those that do not apply)
    - [x] CHANGELOG.md
    - [x] Development documentation
    - [x] User documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
    - [x] Unit tests
- [x] New and existing unit tests pass locally with my changes
    - [x] Automated tests passing (if not fork)
    - [x] Manual testing/integration testing passes (if forked)
- [x] Any dependent changes have been merged and published in downstream modules
- [x] My code has passed security scanning
    - [x] Snyk
    - [x] git-secrets

### CODE REVIEW

File: graphql/src/entities/internal_reconcile_report.py [hunk 1]
```diff
@@ -19,7 +19,7 @@ class Phantom(pydantic.BaseModel):
     key_path: str
     orca_etag: str
     # Python doesn't cap 32 bit/4 byte int size, but GraphQL can't handle larger ints.
-    orca_last_update: float
+    orca_granule_last_update: float
```
Comment (AndrewDorn) [inline graphql/src/entities/internal_reconcile_report.py:22]: Misname on my part. Found when reviewing Gateway docs.

File: modules/orca/main.tf [hunk 2]
```diff
@@ -1,8 +1,3 @@
-locals {
-  # Used for Load Balancer, EC2 Service, and Container ports. Specifies how GQL will be hosted.
-  deploy_graphql = false
```
Comment (AndrewDorn) [inline modules/orca/main.tf:3]: Now that I'm documenting things, I needed an output variable to tie to. Seemed like the right time to remove the flag.

Comment (rizbihassan) [reply modules/orca/main.tf:3]: so this means we will be deploying graphql from now on by default

File: website/docs/developer/api/api.md [hunk 3]
```diff
@@ -10,13 +10,16 @@ The purpose of this page is to give developers information on how to use the ORC
 All ORCA APIs use the `POST` method.
 All API endpoints use AWS IAM authorization.
 
+When using the Load Balancer endpoint for GraphQL, encode your query as a JSON string, and store it in the request body.
```
Comment (rizbihassan) [inline website/docs/developer/api/api.md:13]: might be helpful to show an example of how to do this in request body.

File: website/docs/developer/api/api.md [hunk 4]
```diff
@@ -417,7 +511,7 @@ The following table lists the fields in the output:
 The API returns status code 200 on success, 400 if `jobId` or `pageIndex` are missing and 500 if an error occurs.
 
 ## Internal Reconcile report mismatch API
-The `orca/datamanagement/reconciliation/internal/jobs/job/{jobid}/mismatches` API call receives job id and page index from end user and returns reporting information of files that are missing from ORCA S3 bucket or have different metadata values than what is expected. Note that `{jobid}` is optional.
+The `orca/datamanagement/reconciliation/internal/jobs/job/{jobid}/mismatches` API call receives job id and page index from end user and returns reporting information of files that are missing from ORCA S3 bucket or have different metadata values than what is expected.
```
Comment (rizbihassan) [inline website/docs/developer/api/api.md:514]: Is jobid no longer optional?

**Review by rizbihassan — CHANGES_REQUESTED** [review]

File: website/docs/developer/api/api.md (hunk 4 above)
Comment (AndrewDorn) [reply website/docs/developer/api/api.md:514]: It was optional for requesting the status of a job, but I don't believe it was ever optional for retrieving reports. Likely a copy/paste issue when replicating the job-status docs.

File: website/docs/developer/api/api.md (hunk 3 above)
Comment (AndrewDorn) [reply website/docs/developer/api/api.md]: On second thought, you're right. Even if limited, it's good to have a concrete example. Added.

**Pull request merged**